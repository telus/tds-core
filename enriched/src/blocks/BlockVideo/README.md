BlockVideo example:

    <BlockVideo 
        caption="Never miss a call" 
        subtext="Connect with your team and customers to easily manage calls, host audio and video conferences, and more with this cloud-based business VoIP solution."
        videoUrl="https://www.youtube.com/embed/Mt-v4L5hWEs?cc_load_policy=2"
    />
