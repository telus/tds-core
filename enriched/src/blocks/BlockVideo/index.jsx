import PropTypes from 'prop-types';
import React from 'react';
import './b-video.scss';

/**
 * BlockVideo is composed of a youtube video in an iframe, and optionally, a title and short description underneath..
 * The video is left aligned, taking up 8 grid columns at medium width and higher. When the screen size is smaller, the video is centered.
 *
 * Contentful models (BUS > Marketing): Video Block
 *
 * https://telusdigital.atlassian.net/browse/BMK-55
 */
const BlockVideo = ({ caption, subtext, videoUrl }) => (
  <section className="b-video">
    <div className="container container--limited-width">
      <div className="grid-row">
        <div className="medium-9">
          <div className="b-video__container">
            <iframe allowFullScreen="" frameBorder="0" src={videoUrl} title={caption}/>
          </div>
          { caption && <p className="text text--medium">
            <strong>{caption}: </strong>
            {subtext}
          </p>}
        </div>
      </div>
    </div>
  </section>
);

BlockVideo.propTypes = {
  caption: PropTypes.string,
  subtext: PropTypes.string,
  videoUrl: PropTypes.string.isRequired
};

BlockVideo.defaultProps = {
  subtext: '',
  caption: ''
};

export default BlockVideo;
