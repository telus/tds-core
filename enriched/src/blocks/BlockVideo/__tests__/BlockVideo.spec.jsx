import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import BlockVideo from '../';

describe('BlockVideo', () => {
  const data = { caption: 'Foo', subtext: 'Bar', videoUrl: 'https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1' };

  it('matches snapshot', () => {
    const wrapper = mount(<BlockVideo {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render caption', () => {
    const wrapper = mount(<BlockVideo {...data} />);
    expect(wrapper.find('strong')).toBeDefined();
    expect(wrapper.find('strong').text()).toBe('Foo: ');
  });

  it('should render subtext', () => {
    const wrapper = mount(<BlockVideo {...data} />);
    expect(wrapper.find('p').first().text()).toContain('Bar');
  });

  it('should render without title or subtext', () => {
    const wrapper = mount(<BlockVideo videoUrl={data.videoUrl} />);
    expect(wrapper.find('.text--medium').exists()).toEqual(false);
  });

  it('should render an iframe with video', () => {
    const wrapper = mount(<BlockVideo videoUrl={data.videoUrl} />);
    expect(wrapper.find('iframe').first().prop('src')).toEqual('https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1');
  });
});
