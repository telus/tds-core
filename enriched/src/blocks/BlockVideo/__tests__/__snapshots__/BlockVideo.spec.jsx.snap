exports[`BlockVideo matches snapshot 1`] = `
<BlockVideo
  caption="Foo"
  subtext="Bar"
  videoUrl="https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1">
  <section
    className="b-video">
    <div
      className="container container--limited-width">
      <div
        className="grid-row">
        <div
          className="medium-9">
          <div
            className="b-video__container">
            <iframe
              allowFullScreen=""
              frameBorder="0"
              src="https://www.youtube.com/embed/MaCZN2N6Q_I?cc_load_policy=1"
              title="Foo" />
          </div>
          <p
            className="text text--medium">
            <strong>
              Foo
              : 
            </strong>
            Bar
          </p>
        </div>
      </div>
    </div>
  </section>
</BlockVideo>
`;
